### PR TITLE
`SelectionUIService` move contained classes/struct into own files

### PR DESCRIPTION
**Pull request by elachlan**
Make `SelectionUIService` partial and move contained classes to own files:
- SelectionUIService.ContainerSelectionUIItem.cs
- SelectionUIService.HitTestInfo.cs
- SelectionUIService.SelectionUIItem.cs

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/8321)

### CODE REVIEW

**Comment by elachlan** [comment]
Flaky tests.

**Review by lonitra — APPROVED** [review]

File: src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/SelectionUIService.ContainerSelectionUIItem.cs [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+                    return null;
+                }
+            }
+
+            public override int GetHitTest(Point pt)
+            {
+                int ht = NOHIT;
+                if ((GetRules() & SelectionRules.Visible) != SelectionRules.None && !_outerRect.IsEmpty)
+                {
+                    Rectangle r = new Rectangle(_outerRect.X, _outerRect.Y, CONTAINER_WIDTH, CONTAINER_HEIGHT);
```
Comment (dreddy-work) [inline src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/SelectionUIService.ContainerSelectionUIItem.cs:37]:
```suggestion
                    Rectangle r = new(_outerRect.X, _outerRect.Y, CONTAINER_WIDTH, CONTAINER_HEIGHT);
```

File: src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/SelectionUIService.ContainerSelectionUIItem.cs [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+                else
+                {
+                    return null;
+                }
+            }
+
+            public override int GetHitTest(Point pt)
+            {
+                int ht = NOHIT;
+                if ((GetRules() & SelectionRules.Visible) != SelectionRules.None && !_outerRect.IsEmpty)
```
Comment (dreddy-work) [inline src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/SelectionUIService.ContainerSelectionUIItem.cs:35]: nit: reduce nesting by inverting this

File: src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/SelectionUIService.ContainerSelectionUIItem.cs [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+            public override void DoPaint(Graphics gr)
+            {
+                // If we're not visible, then there's nothing to do...
+                if ((GetRules() & SelectionRules.Visible) == SelectionRules.None)
+                {
+                    return;
+                }
+
+                Rectangle glyphBounds = new Rectangle(_outerRect.X, _outerRect.Y, CONTAINER_WIDTH, CONTAINER_HEIGHT);
```
Comment (dreddy-work) [inline src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/SelectionUIService.ContainerSelectionUIItem.cs:60]:
```suggestion
                Rectangle glyphBounds = new(_outerRect.X, _outerRect.Y, CONTAINER_WIDTH, CONTAINER_HEIGHT);
```

Comment (dreddy-work) [reply src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/SelectionUIService.ContainerSelectionUIItem.cs:60]: and other places in this file.

File: src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/SelectionUIService.HitTestInfo.cs [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+        {
+            public readonly int hitTest;
+            public readonly SelectionUIItem selectionUIHit;
+            public readonly bool containerSelector;
+
+            public HitTestInfo(int hitTest, SelectionUIItem selectionUIHit)
+            {
+                this.hitTest = hitTest;
+                this.selectionUIHit = selectionUIHit;
+                containerSelector = false;
```
Comment (dreddy-work) [inline src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/SelectionUIService.HitTestInfo.cs:19]: `false` is default. do you need this assignment?

File: src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/SelectionUIService.HitTestInfo.cs [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+
+            public bool Equals(HitTestInfo other)
+                => hitTest == other.hitTest
+                    && selectionUIHit == other.selectionUIHit
+                    && containerSelector == other.containerSelector;
+
+            public static bool operator ==(HitTestInfo left, HitTestInfo right)
+            {
+                return left.Equals(right);
+            }
```
Comment (dreddy-work) [inline src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/SelectionUIService.HitTestInfo.cs:56]:
```suggestion
            public static bool operator ==(HitTestInfo left, HitTestInfo right) => left.Equals(right);
```

File: src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/SelectionUIService.SelectionUIItem.cs [hunk 6]
```diff
@@ -0,0 +1,542 @@
...
+
+                bool fActive = false;
+                if (_selUIsvc._selSvc is not null)
+                {
+                    fActive = _component == _selUIsvc._selSvc.PrimarySelection;
+                    // Office rules:  If this is a multi-select, reverse the colors for active / inactive.
+                    fActive = (fActive == (_selUIsvc._selSvc.SelectionCount <= 1));
+                }
+
+                Rectangle r = new Rectangle(_outerRect.X, _outerRect.Y, GRABHANDLE_WIDTH, GRABHANDLE_HEIGHT);
```
Comment (dreddy-work) [inline src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/SelectionUIService.SelectionUIItem.cs:136]:
```suggestion
                Rectangle r = new(_outerRect.X, _outerRect.Y, GRABHANDLE_WIDTH, GRABHANDLE_HEIGHT);
```

File: src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/SelectionUIService.SelectionUIItem.cs [hunk 7]
```diff
@@ -0,0 +1,542 @@
...
+                Rectangle inner = _innerRect;
+                Rectangle outer = _outerRect;
+                Region oldClip = gr.Clip;
+                Color borderColor = SystemColors.Control;
+                if (_control is not null && _control.Parent is not null)
```
Comment (dreddy-work) [inline src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/SelectionUIService.SelectionUIItem.cs:141]:
```suggestion
                if (_control?.Parent is not null)
```

File: src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/SelectionUIService.SelectionUIItem.cs [hunk 8]
```diff
@@ -0,0 +1,542 @@
...
+                // Is it within our rects?
+                if (!PointWithinSelection(pt))
+                {
+                    return NOHIT;
+                }
+
+                // Which index in the array is this?
+                int nOffset = GetHandleIndexOfPoint(pt);
+                // If no index, the user has picked on the hatch
+                if (-1 == nOffset || _sizes[nOffset] == 0)
```
Comment (dreddy-work) [inline src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/SelectionUIService.SelectionUIItem.cs:235]: nit: same here.

**Comment by elachlan** [comment]
@dreddy-work this PR moves the classes into their own files. I'd like to keep that commit clean of any refactoring. I will be coming back through with another PR with the HashTable changes.

**Review by dreddy-work — APPROVED** [review]

File: src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/SelectionUIService.SelectionUIItem.cs (hunk 8 above)
Comment (elachlan) [reply src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/SelectionUIService.SelectionUIItem.cs:235]: I believe stylecop has an analyzer for yoda conditions. I will put in a PR to enable it.

**Pull request merged**